src/components/List/List.tsx

```diff
@@ -98,20 +100,32 @@ export class List<T = unknown> extends React.Component<ListProps<T>, ListState<T
         if (this.props.activeItemIndex !== prevProps.activeItemIndex) {
             this.activateItem(this.props.activeItemIndex);
         }
+
+        if (this.props.onChangeActive && this.state.activeItem !== prevState.activeItem) {
```

Comment:
There is a better place for such logic - https://github.com/gravity-ui/uikit/blob/d51039bb0fd4129ac511681e7fa23f051d1c3397/src/components/List/List.tsx#L157

Reply:
The function activateItem is not the only place where this.state.activeItem is modified. It is also modified, for example, in the [deactivate](https://github.com/gravity-ui/uikit/blob/d51039bb0fd4129ac511681e7fa23f051d1c3397/src/components/List/List.tsx#L384) function  and [onSortEnd](https://github.com/gravity-ui/uikit/blob/d51039bb0fd4129ac511681e7fa23f051d1c3397/src/components/List/List.tsx#L470). I don't want to duplicate the logic in each of these places, so I have moved it to componentDidUpdate.